### PR TITLE
replace httpResponse with SdkHttpResponse

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
Replacing `HttpResponse` with `SdkHttpResponse`

## Testing
Unit tests and integration tests succeeded.

will add changelog entry once approved

### CODE REVIEW

File: core/aws-core/src/test/java/software/amazon/awssdk/awscore/client/http/StaxResponseHandlerComponentTest.java [hunk 1]
```diff
@@ -68,8 +69,9 @@ public void saxParserShouldNotExposeLocalFileSystem() throws Exception {
 
         StaxResponseHandler<EmptyAwsResponse> responseHandler = new StaxResponseHandler<>(dummyUnmarshaller());
 
-        HttpResponse response = new HttpResponse(ValidSdkObjects.sdkHttpFullRequest().build());
-        response.setContent(new ByteArrayInputStream(payload.getBytes(Charset.forName("UTF-8"))));
+        SdkHttpFullResponse response = ValidSdkObjects.sdkHttpFullResponse()
+                                                      .content(new AbortableInputStream(new ByteArrayInputStream(payload.getBytes(Charset.forName("UTF-8"))), () -> {}))
```
Comment (millems) [inline core/aws-core/src/test/java/software/amazon/awssdk/awscore/client/http/StaxResponseHandlerComponentTest.java]: StandardCharsets.UTF_8?

File: http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpResponse.java [hunk 2]
```diff
@@ -39,4 +39,11 @@
      * <p>This will always be positive.</p>
      */
     int statusCode();
+
+    /**
+     * If we get back any 2xx status code, then we know we should treat the service call as successful.
+     */
+    default boolean isSuccesssfull() {
```
[the remark this answers is not in the document]
Comment (zoewangg) [reply http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpResponse.java]: Good catch! The response clearly wants to be extra successful 😛

File: core/aws-core/src/main/java/software/amazon/awssdk/awscore/eventstream/EventStreamAsyncResponseTransformer.java [hunk 3]
```diff
@@ -234,15 +239,23 @@ private boolean isException(Message m) {
...
+    private SdkHttpFullResponse adaptMessageToResponse(Message message) {
+
+        Map<String, List<String>> headers =
+            message.getHeaders()
+                   .entrySet()
+                   .stream()
+                   .collect(HashMap::new, (m, e) -> m.put(e.getKey(), singletonList(e.getValue().getString())), Map::putAll);
+
+        return SdkHttpFullResponse.builder()
+                                  .content(new AbortableInputStream(new ByteArrayInputStream(message.getPayload()), () -> {
```
Comment (shorea) [inline core/aws-core/src/main/java/software/amazon/awssdk/awscore/eventstream/EventStreamAsyncResponseTransformer.java]: Can we have a constructor for AbortableInputStream that just takes an input stream? Or perhaps a create method instead.

Comment (zoewangg) [reply core/aws-core/src/main/java/software/amazon/awssdk/awscore/eventstream/EventStreamAsyncResponseTransformer.java]: +1 will add

**Pull request merged**